runtime/lua/vim/lsp/buf.lua

```diff
@@ -15,18 +15,52 @@ local function npcall(fn, ...)
   return ok_or_nil(pcall(fn, ...))
 end
 
-local function request(method, params, callback)
+local function request(method, default_params, overrides)
```

Comment:
I might want to expose or change where this is defined, so that it's easy to extend with your own request style thing that might do this?